src/main/java/conan/ui/configuration/ConanConfig.java

```diff
@@ -72,17 +72,14 @@ public String getDisplayName() {
     @Override
     public JComponent createComponent() {
         downloadConfiguration.addActionListener(actionEvent -> {
-            String url = configInstallUrl.getText();
-            if (StringUtils.isBlank(url)) {
-                url = PersistencyUtils.getValue(CONFIG_INSTALL_URL);
+            String location = configInstallLocation.getText();
+            if (StringUtils.isBlank(location)) {
+                location = PersistencyUtils.getValue(CONFIG_INSTALL_LOCATION);
             }
-            PersistencyUtils.setValue(CONFIG_INSTALL_URL, url);
-            if (!Utils.validateUrl(url)) {
-                setConfigInstallRes("Bad URL", false);
-                return;
-            }
-            runConfigInstall(url);
+            PersistencyUtils.setValue(CONFIG_INSTALL_LOCATION, location);
+            runConfigInstall(location);
         });
+        configInstallLocation.getEmptyText().setText("Directory or URL");
```

Comment:
```suggestion
        configInstallLocation.getEmptyText().setText("Git repository, local folder or ZIP file (local or http)");
```


Maybe it is too long, but it is what we currently have in the help message

Reply:
Looking here: https://docs.conan.io/en/latest/reference/commands/consumer/config.html

We might want to provide a checkbox to add `--type git` to the config install command.
> You can also force the git download by using --type git (in case it is not deduced from the URL automatically):

Reply:
The above requires the checkbox to be persisted.

I'm not currently comfotable making changes to the persistency because we don't seem to have any unit tests for it.

I might make an issue for this force_git thing, and come back to it after we have some more robust testing. What do you think?

Reply:
Yes, don't worry about that. IMHO all the config should be refactored in the future, t could be better to have a submenu for the configuration populated by [all] the values in `.conan/conan.con` and `.conan/remotes.json` and let the user modify them individually (these values would be persisted by Conan itself, not by CLion)... and the download/install button would be just an action, and at that moment it will request a URL/file.

It is a big change, I'm not sure about it, but it would make the integration much better if the user doesn't need to go to the command line to do some modifications.

Reply:
I like the sound of that. :+1:

Reply:
The only problem is Conan changing the format of those files. It is easier to parse/write the files than to use json output from CLI interface. We would need to abstract file format and implement reader/writers for different Conan versions.